### PR TITLE
Update dist-s390x-dist image

### LINKED ISSUE #94324: Provide rust-lld on s390x (for the wasm target)
Trying to build for the `wasm32-unknown-unknown` target on a s390x host fails with:
```
  `rust-lld` not found, please install it!
```

This seems to be because - unlike on other architectures - `rust-lld` is not built and packaged as part of the rust toolchain.

Now, it is true that LLD currently does not support the s390x architecture.  But that is only an issue when trying to use the s390x architecture as **target** - it is completely fine to build LLD on a s390x host and use it to link binaries for any **other** target architecture it supports, in particular including wasm.

Would it be possible to include `rust-lld` on s390x as well, to make building wasm binaries more straightforward?

### PR DESCRIPTION
**Pull request by nikic**
Update to Ubuntu 20.04 and crosstool-ng 1.24.0. I've upgraded the
ct-ng config and then manually reset the kernel and glibc versions
to the oldest supported.

Specifically, we're updating from kernel 2.6.32.68 to 2.6.32.71
and glibc 2.11.1 to 2.12.1 here. The compiler toolchain is also
updated, but I don't think that's relevant for compatibility.

I've also enabled LLD, so this fixes #94324.

r? @Mark-Simulacrum cc @uweigand

### CODE REVIEW

**Comment by uweigand** [comment]
Many thanks for addressing this so quickly!

The move from glibc 2.11 to 2.12 should be fine.  The only distribution theoretically excluded by that is SLES11, which will go out of even long-term support next month, so this shouldn't be an issue.

File: src/ci/docker/README.md [hunk 1]
```diff
@@ -271,8 +271,8 @@ For targets: `s390x-unknown-linux-gnu`
 - Target options > Target Architecture = s390
 - Target options > Bitness = 64-bit
 - Operating System > Target OS = linux
-- Operating System > Linux kernel version = 2.6.32.68 -- ~RHEL6 kernel
-- C-library > glibc version = 2.11.1 -- ~SLE11-SP4 glibc
-- C compiler > gcc version = 5.2.0
+- Operating System > Linux kernel version = 2.6.32.71 -- ~RHEL6 kernel
+- C-library > glibc version = 2.12.1 -- ~SLE11-SP4 glibc
```
Comment (cuviper) [inline src/ci/docker/README.md:275]:
```suggestion
- C-library > glibc version = 2.12.1 -- ~RHEL6 glibc
```

**Comment by Mark-Simulacrum** [comment]
Can you also upgrade the platform support doc? "S390x Linux (kernel 2.6.32, glibc 2.11)" is the current listing, presumably the latter should be slightly bumped.

**Comment by nikic** [comment]
@Mark-Simulacrum Done!

**Pull request merged**